src/huntsman/pocs/states/huntsman/coarse_focusing.py

```diff
@@ -27,24 +24,48 @@ def on_enter(event_data):
 
     # Convert altaz coordinates to radec.
     coarse_focus_coords = altaz_to_radec(alt=coarse_focus_alt, az=coarse_focus_az,
-                                         location=earth_location, obstime=current_time())
+                                         location=pocs.observatory.earth_location,
+                                         obstime=current_time())
 
     pocs.say(f'Coarse focus coordinates: {coarse_focus_coords}')
 
     # Slew to coordinates for coarse focusing.
     pocs.say("I'm slewing over to the coordinates for coarse focus.")
     pocs.observatory.mount.slew_to_coordinates(coarse_focus_coords)
-
     pocs.say("I'm at the coordinates for coarse focus.")
 
-    # Do the autofocusing
-    pocs.say("Coarse focusing all cameras.")
+    # Do the autofocusing (non-blocking).
+    pocs.say("Starting coarse focus on all cameras.")
     autofocus_events = pocs.observatory.autofocus_cameras(coarse=True)
+
+    timer = CountdownTimer(coarse_focus_timeout)
     pocs.logger.debug("Waiting for coarse focus to finish.")
-    wait_for_events(list(autofocus_events.values()), timeout=coarse_focus_timeout)
 
-    # Morning and not dark enough for observing...
-    if pocs.observatory.past_midnight and not pocs.is_dark(horizon='observe'):
-        pocs.next_state = 'twilight_flat_fielding'
-    else:
-        pocs.next_state = 'scheduling'
+    while True:
+        # Check if all cameras are finished and proceed if so.
+        if all([e.is_set() for e in autofocus_events.values()]):
```

Comment:
We are still using events for pyro autofocus, but at some stage we should switch to `concurrent.futures`, which implements a nice `wait` function (https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.wait) that will all us to use the timeout intrinsic to the focuser, which is a bit cleaner.